app/src/main/java/ghcharactertracker/com/ghcharactertracker/ScenarioActivity.java

```diff
@@ -0,0 +1,231 @@
+        healthText = (EditText) findViewById(R.id.scenarioHealth);
+        expText = (EditText) findViewById(R.id.scenarioExp);
+        moneyText = (EditText) findViewById(R.id.scenarioMoney);
+
+        dbHandler = DBHandler.getDbHandler(this);
+
+        addTextWatchers();
+        updateUI();
+    }
+
+    public void changeHealth(View v) {
+        switch(v.getId()){
+            case R.id.healthAdd:
+                updateModelScenarioHealth(scenario.getHealth() + 1);
+                break;
+            case R.id.healthSub:
+                updateModelScenarioHealth(scenario.getHealth() - 1);
+                break;
+        }
+
+        updateHealthUI();
+    }
+
+    public void changeExp(View v) {
+        switch(v.getId()){
+            case R.id.expAdd:
+                updateModelScenarioExp(scenario.getExp() + 1);
+                break;
+            case R.id.expSub:
+                updateModelScenarioExp(scenario.getExp() - 1);
+                break;
+        }
+
+        updateExpUI();
+    }
+
+    public void changeMoney(View v) {
+        switch(v.getId()){
+            case R.id.moneyAdd:
+                updateModelScenarioMoney(scenario.getMoneyTokens() + 1);
+                break;
+            case R.id.moneySub:
+                updateModelScenarioMoney(scenario.getMoneyTokens() - 1);
+                break;
+        }
+
+        updateMoneyUI();
+    }
+
+    private void updateUI() {
+        updateHealthUI();
+        updateExpUI();
+        updateMoneyUI();
+        updateLevelUI();
+    }
+
+    private void updateHealthUI() {
+        healthText.setText(Integer.toString(scenario.getHealth()));
+    }
+
+    private void updateExpUI() {
+        expText.setText(Integer.toString(scenario.getExp()));
+    }
+
+    private void updateMoneyUI() {
+        moneyText.setText(Integer.toString(scenario.getMoneyTokens()));
+    }
+
+    private void updateLevelUI() {
+        if (scenario.getLevel() > 0) {
+            levelText.setText(Integer.toString(scenario.getLevel()));
+        }
+    }
+
+    private void updateModelScenarioLevel(int level) {
+        if(scenario.getLevel() != level) {
+            scenario.setLevel(level);
+            dbHandler.updateScenario(scenario);
+        }
+    }
+
+    private void updateModelScenarioHealth(int health) {
+        if(scenario.getHealth() != health) {
+            scenario.setHealth(health);
+            dbHandler.updateScenario(scenario);
+        }
+    }
+
+    private void updateModelScenarioExp(int exp) {
+        if(scenario.getExp() != exp) {
+            scenario.setExp(exp);
+            dbHandler.updateScenario(scenario);
+        }
+    }
+
+    private void updateModelScenarioMoney(int money) {
+        if(scenario.getMoneyTokens() != money) {
+            scenario.setMoneyTokens(money);
+            dbHandler.updateScenario(scenario);
+        }
+    }
+
+    private void addTextWatchers() {
+        levelText.addTextChangedListener(new TextWatcher() {
```

Comment:
There's probably some clever way to simplify the repeated logic for adding the TextChangedListeners by using a `Callable`, but probably not worth it and I don't want to think about it 💃

Reply:
Maybe it'd be easy to subclass TextChangedListener and put in the empty overrides there so you didn't have to repeat that at least? Again, probably not worth it.